src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,41 @@
+package duke.command;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+import duke.TaskList;
+import duke.Ui;
+import duke.exception.InvalidInputException;
+import duke.exception.InvalidInputException.InputExceptionType;
+import duke.exception.SaveException;
+import duke.task.Task;
+
+// Delete a task from the task list
+public class DeleteCommand extends Command {
+    public DeleteCommand(Ui ui, TaskList tasks, HashMap<String, String> arguments) {
+        super(ui, tasks, arguments);
+    }
+
+    @Override
+    public void execute() throws InvalidInputException, IOException, SaveException {
+        String indexString = arguments.get("payload");
+        if (indexString == null) {
+            // An index must be provided for the task to be marked "done"
+            throw new InvalidInputException(InputExceptionType.EMPTY_INDEX);
+        } else {
```

Comment:
You can consider removing this else clause so that there's lesser tendency for your code to be deeply nested

Reply:
Thanks for the advice. This has been fixed in the latest commit.